src/routes/api/assets/getById.js

```diff
@@ -1,16 +1,50 @@
 const logger = require("../../../logger");
+const { z } = require("zod");
+const { PrismaClientKnownRequestError } = require("@prisma/client").Prisma;
+const asset = require("../../../model/asset");
 const { createSuccessResponse } = require("../../../response");
 
 /**
- * Get an asset by it's id
+ * Get an asset by its id
  */
-
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
   logger.debug(
-    { user: req.user, id: req.params.id },
+    { user: req.user, id: req.params.assetId },
     `received request: GET /v1/assets/:assetId`,
   );
 
-  return res.status(418).json(createSuccessResponse());
+  // Validate UUID format
+  try {
+    z.string().uuid().parse(req.params.assetId);
+  } catch (error) {
+    logger.debug({ error }, "invalid asset ID format");
+    return next({ status: 400, message: "Invalid uuid" });
+  }
+
+  // Get asset using the model's get function
+  let foundAsset;
+  try {
+    foundAsset = await asset.get(req.params.assetId);
+  } catch (error) {
+    logger.error({ error }, "Error fetching asset");
+
+    if (
+      error instanceof PrismaClientKnownRequestError &&
+      error.code === "P2025"
+    ) {
+      return next({ status: 404, message: "Asset not found" });
+    }
+    return next({ status: 500, message: "Internal server error" });
+  }
+
+  // Check visibility permissions
+  if (
+    foundAsset.visibility !== "public" &&
+    foundAsset.creatorId !== req.user?.id
```

Comment:
issue: creatorId is the user's hashed email, so we can compare it against `req.user`

```suggestion
    foundAsset.creatorId !== req.user
```